deployment/ccip/changeset/cs_deploy_token_pools.go

```diff
@@ -0,0 +1,497 @@
+	TokenAdminRegistry *token_admin_registry.TokenAdminRegistry
+	TokenPool          *token_pool.TokenPool
+	TokenAddress       common.Address
+	TimelockAddress    common.Address
+	ExternalAdmin      common.Address
+	RegistryState      token_admin_registry.TokenAdminRegistryTokenConfig
+	RemoteChainsToAdd  RemoteChains
+	OwnedByTimelock    bool
+}
+
+// RateLimiterConfig defines the inbound and outbound rate limits for a remote chain.
+type RateLimiterConfig struct {
+	Inbound  token_pool.RateLimiterConfig
+	Outbound token_pool.RateLimiterConfig
+}
+
+// validateRateLimterConfig validates rate and capacity in accordance with on-chain code.
+// see https://github.com/smartcontractkit/ccip/blob/ccip-develop/contracts/src/v0.8/ccip/libraries/RateLimiter.sol.
+func validateRateLimiterConfig(rateLimiterConfig token_pool.RateLimiterConfig) error {
+	if rateLimiterConfig.IsEnabled {
+		if rateLimiterConfig.Rate.Cmp(rateLimiterConfig.Capacity) >= 0 || rateLimiterConfig.Rate.Cmp(zero()) == 0 {
+			return errors.New("rate must be greater than 0 and less than capacity")
+		}
+	} else {
+		if rateLimiterConfig.Rate.Cmp(zero()) != 0 || rateLimiterConfig.Capacity.Cmp(zero()) != 0 {
+			return errors.New("rate and capacity must be 0")
+		}
+	}
+	return nil
+}
+
+// RemoteChainsConfig defines rate limits for remote chains.
+type RemoteChains map[uint64]RateLimiterConfig
+
+func (rc RemoteChains) Validate() error {
+	for chainSelector, chainConfig := range rc {
+		if err := validateRateLimiterConfig(chainConfig.Inbound); err != nil {
+			return fmt.Errorf("validation of inbound rate limiter config for remote chain with selector %d failed: %w", chainSelector, err)
+		}
+		if err := validateRateLimiterConfig(chainConfig.Outbound); err != nil {
+			return fmt.Errorf("validation of outbound rate limiter config for remote chain with selector %d failed: %w", chainSelector, err)
+		}
+	}
+	return nil
+}
+
+// BaseTokenPoolInput defines all the information required of the user to configure new and existing pools.
+type BaseTokenPoolInput struct {
+	RemoteChainsToAdd RemoteChains
+	TokenAddress      common.Address
+	ExternalAdmin     common.Address
+}
+
+func (t BaseTokenPoolInput) Validate() error {
+	if err := t.RemoteChainsToAdd.Validate(); err != nil {
+		return fmt.Errorf("failed to validate remote chains config: %w", err)
+	}
+	return nil
+}
+
+// NewTokenPoolInput defines all information required of the user to deploy and configure a new token pool.
+type NewTokenPoolInput struct {
+	BaseTokenPoolInput
+	Type               deployment.ContractType
+	LocalTokenDecimals uint8
+	AllowList          []common.Address
+	AcceptLiquidity    *bool
+}
+
+func (t NewTokenPoolInput) Validate() error {
+	if err := t.RemoteChainsToAdd.Validate(); err != nil {
+		return fmt.Errorf("failed to validate remote chains config: %w", err)
+	}
+	if t.Type != BurnMintTokenPool && t.Type != BurnFromMintTokenPool && t.Type != BurnWithFromMintTokenPool && t.Type != LockReleaseTokenPool {
```

Comment:
Let's have a map with acceptable tokenpool type. something like
```
map[TypeAndVersion]struct{}{}
```
you can then add support for more if required